geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/SftArgResolver.scala

```diff
@@ -22,84 +23,100 @@ import scala.util.{Failure, Success, Try}
-        case Success(sft) => Some(sft)
-        case Failure(e) =>
-          logger.debug(s"Unable to parse sft spec from string $specArg with error ${e.getMessage}")
-          None
-      }
+  private[SftArgResolver] def parseSpecStringFile(args: argTuple): resEither =
+    Try(SimpleFeatureTypes.createType(
+      Option(args._2).getOrElse(throw new Throwable("Null feature name provided. " +
+        "Unable to parse spec string from null value.")),
+      FileUtils.readFileToString(new File(Option(args._1).getOrElse(""))))
+    ) match {
+      case Success(sft) => Right(sft)
+      case Failure(e) => Left((s"Unable to parse sft spec from string ${args._1} with error ${e.getMessage}", e, PATH))
     }
 
-  private[SftArgResolver] def parseConf(configStr: String, name: String): Option[SimpleFeatureType] =
+  private[SftArgResolver] def parseConf(configStr: String, name: String): Either[Throwable, SimpleFeatureType] = {
     Try {
       val sfts = SimpleSftParser.parseConf(ConfigFactory.parseString(configStr, parseOpts))
       if (sfts.size > 1) logger.warn(s"Found more than one SFT conf in arg '$configStr'")
       sfts.get(0)
     } match {
-      case Success(sft) if name == null || name == sft.getTypeName => Some(sft)
-      case Success(sft) => Some(SimpleFeatureTypes.renameSft(sft, name))
-      case Failure(e) => throw e
+      case Success(sft) if name == null || name == sft.getTypeName => Right(sft)
+      case Success(sft) => Right(SimpleFeatureTypes.renameSft(sft, name))
+      case Failure(e) => Left(e)
     }
+  }
 
   // gets an sft based on a spec conf string
-  private[SftArgResolver] def parseConfStr(specArg: String, name: String): Option[SimpleFeatureType] =
-    Try(parseConf(specArg, name)) match {
-      case Success(sftOpt) => sftOpt
-      case Failure(e) =>
-        logger.debug(s"Unable to parse sft spec from string $specArg as conf with error ${e.getMessage}")
-        None
+  private[SftArgResolver] def parseConfStr(args: argTuple): resEither =
+    parseConf(args._1, args._2) match {
+      case Right(sftOpt) => Right(sftOpt)
+      case Left(e) => Left((s"Unable to parse sft spec from string ${args._1} as conf.", e, CONFSTR))
     }
 
   // parse spec conf file
-  private[SftArgResolver] def parseConfFile(specArg: String, name: String): Option[SimpleFeatureType] = {
-    Try(parseConf(FileUtils.readFileToString(new File(specArg)), name)) match {
-      case Success(sftOpt) => sftOpt
-      case Failure(e) =>
-        logger.debug(s"Unable to parse sft spec from file $specArg as conf with error ${e.getMessage}")
-        None
+  private[SftArgResolver] def parseConfFile(args: argTuple): resEither =
+    Try {
+      FileUtils.readFileToString(new File(args._1))
+    } match {
+      case Success(file) => parseConf(file, args._2) match {
+          case Right(sftOpt) => Right(sftOpt)
+          case Left(e) => Left((s"Unable to parse sft spec from file ${args._1} as conf.", e, PATH))
```

Comment:
[minor] i think you could use Either.left here instead of the match - that lets you modify the Left if it's a Left but keep the Right unchanged if it's a Right:
`parseConf(file, args._2).left.map(e => (s"Unable to parse sft spec from file ${args._1} as conf.", e, PATH))
there's no functional difference, but that's the idiomatic way to use Either